### PR TITLE
added server selection

### PR DESCRIPTION
**Pull request by steffli**
Hey Ming,

i cleaned up some messy coding of mine and added a server dropdown to choose between De-Tue and USA-UCR referencing to the corresponding urls.
Created an external function which takes the button, the urls as well as parameters and selected server. I changed the layout and callback functions accordingly.

Best,
Stephan

### CODE REVIEW

**Comment by mwang87** [comment]
Awesome, this looks good. We'll not merge it just quite yet, until the DE server is up and running. Then we can merge and deploy!

File: Dockerfile [hunk 1]
```diff
@@ -1,3 +1,13 @@
+FROM continuumio/miniconda3:4.10.3 as cleanup
+
+# Set the container name
+
+# Stop and remove the container, delete the image
```
Comment (mwang87) [inline Dockerfile:5]: Can we remove these new parts as they shouldnt be in the docker compose

File: __pycache__/app.cpython-39.pyc
Comment (mwang87) [inline __pycache__/app.cpython-39.pyc:1]: Please remove all these pyc files.

File: app.py [hunk 2]
```diff
@@ -64,6 +70,22 @@
     dark=False,
     sticky="top",
 )
+# server_dropdown_1 = dcc.Dropdown(
```
Comment (mwang87) [inline app.py:73]: go ahead and remove these commented sections.

File: app.py [hunk 3]
```diff
@@ -405,19 +443,36 @@ def _determine_row_selection_list(file_table_data, selected_table_data, get_all=
...
               ],
-              [
+              [   
                   Input('dataset_accession', 'value'), 
                   Input('dataset_password', 'value'), 
                   Input('file-table', 'derived_virtual_data'),
                   Input('file-table', 'derived_virtual_selected_rows'),
                   Input('file-table2', 'derived_virtual_data'),
                   Input('file-table2', 'derived_virtual_selected_rows'),
+                  Input('server-dropdown', 'value'),
```
Comment (mwang87) [inline app.py:469]: are two fields necessary?

File: app.py [hunk 4]
```diff
@@ -427,15 +482,21 @@ def create_link(accession, dataset_password, file_table_data, selected_table_dat
...
+    
 
     url_params = {}
     url_params["usi"] = usi_string1
     url_params["usi2"] = usi_string2
 
     total_file_count = len(usi_list1) + len(usi_list2)
 
+
+    dashboard_us = "https://dashboard.gnps2.org/#"
```
Comment (mwang87) [inline app.py:494]: Perhaps make this a dictionary? Then it makes it super easy to add to.

File: app.py [hunk 5]
```diff
@@ -496,34 +532,25 @@ def create_link(accession, dataset_password, file_table_data, selected_table_dat
...
     )
 
-
-    
-
     # Here we will have the GNPS2 analysis urls
-    gnps2_url = "https://gnps2.org/workflowinput?workflowname=classical_networking_workflow"
-
+    gnps2_url = "https://gnps2.org/workflowinput?workflowname=classical_networking_workflow#"
+    gnps2_de = "https://de.gnps2.org/workflowinput?workflowname=classical_networking_workflow#"
```
Comment (mwang87) [inline app.py:537]: This we likely don't need yet.

File: docker-compose.yml
Comment (mwang87) [inline docker-compose.yml:1]: This should be reverted

File: docker-compose.yml.bak
Comment (mwang87) [inline docker-compose.yml.bak:1]: clean up.

File: logs/access.log
Comment (mwang87) [inline logs/access.log:1]: this should not be checked in

File: requests_cache.sqlite
Comment (mwang87) [inline requests_cache.sqlite:1]: not checked in

File: restart.sh
Comment (mwang87) [inline restart.sh:1]: not checked in

File: testing.py
Comment (mwang87) [inline testing.py:1]: not checked in

**Comment by steffli** [comment]
Hey Ming, 
i updated the code and removed the unwanted files. Can you check again if it suits now?

**Pull request merged**